tenable/io/v3/vm/scanners.py

```diff
@@ -0,0 +1,364 @@
+"""
+scanners
+========
+
+The following methods allow for interaction into the Tenable.io
+:devportal:`scanners <scanners>` API.
+
+Methods available on ``tio.v3.vm.scanners``:
+
+.. rst-class:: hide-signature
+.. autoclass:: ScannersAPI
+
+    .. automethod:: control_scan
+    .. automethod:: delete
+    .. automethod:: details
+    .. automethod:: edit
+    .. automethod:: get_aws_targets
+    .. automethod:: get_scanner_key
+    .. automethod:: get_scans
+    .. automethod:: linking_key
+    .. automethod:: search
+    .. automethod:: toggle_link_state
+"""
+# from tenable.io.v3.base import TIOEndpoint
+from tenable.io.v3.base.endpoints.uw import UWBaseEndpoint
+
+
+class ScannersAPI(UWBaseEndpoint):
+    def linking_key(self):
+        """
+        The linking key for the Tenable.io instance.
+
+        Returns:
+            :obj:`str`:
+                The linking key
+
+        Examples:
+            >>> print(tio.v3.vm.scanners.linking_key())
+        """
+        scanners = self.list()
+        for scanner in scanners:
+            if (
+                scanner["uuid"]
+                == "00000000-0000-0000-0000-00000000000000000000000000001"
+            ):
+                return scanner["key"]
+
+    def allowed_scanners(self):
+        """
+        A simple convenience function that returns the list of scanners that the
+        current user is allowed to use.
+
+        Returns:
+            :obj:`list`:
+                List of scanner documents.
+
+        Examples:
+            >>> for scanner in tio.v3.vm.scanners.allowed_scanners():
+            ...     pprint(scanner)
+        """
+        # We want to get the scanners that are available for scanning.  To do so,
+        # we will want to pull the information from the scan template.  This
+        # isn't the prettiest way to handle this, however it will consistently
+        # return the results that we are looking for.
+        def get_scanners(tmpl):
+            for item in tmpl["settings"]["basic"]["inputs"]:
+                if item["id"] == "scanner_id":
+                    return item["options"]
+            return []
+
+        vm_tmpl = self._api.policies.templates().get("advanced", None)
+        was_tmpl = self._api.policies.templates().get("was_scan", None)
```

Comment:
v1 WAS API.  should be deprecated.

Reply:
Updated the method with the following for now.

`return NotImplementedError(
            "This method will be updated once Policies and Editor API's are implemented in v3"
        )`